### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by chambliss**
No big changes here. Summary:
- Small changes/fixes to train_model.py, as discussed - now we should get more accurate results as to whether accuracy is improving! (This includes adding in the confusion matrix and training accuracy, as we talked about on Slack - commit fb36f6e)
- Rewrote the preprocessing for factRuEval-2016 to leverage spaCy's spans to BILUO tags function, so now the preprocessing script is both shorter -and- more accurate than the previous one!
- Updated README
- Preprocessed data for use with spaCy models (they like it in JSON format rather than BILUO tags)
- Fixed git submodule error by removing and re-adding factRuEval-2016 directory (this will make the pull req kind of annoying to scroll through, sorry - recommend clicking on the commit hash for whichever one you're interested in looking at)
- Black and flake8 also ran on train_model.py and combine_ru_data.py (the new script), so they have had formatting changes (looks nice - I'm a fan!)
- Evaluated spaCy's en_core_web_lg model on the combined EN dataset. Performance leaves something to be desired (more detailed notes in the spacy_eval notebook)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,30 @@
-# Multilingual_NER
-Applying BERT to named entity recognition in English and Russian.
```
Comment (ninalopatina) [inline README.md]: Something more like the following:  "This repository applies BERT to named entity recognition in English and Russian in order to improve machine translation quality estimation between English-Russian sentence pairs. Named entities are a known challenge in machine translation, and this repository aims to quantify accuracy by comparing named entity recognition between Russian and English in translated sentence pairs."

File: README.md [hunk 2]
```diff
@@ -1,2 +1,30 @@
...
+# multilingual_NER
+
+## About
+Deployable [BERT](https://github.com/google-research/bert) ensemble models for named entity recognition in English and Russian. `multilingual_NER` provides a toolkit to train, validate, and deploy NER models with an interactive frontend and callable API.
+
+## Installation
```
Comment (ninalopatina) [inline README.md]: Installation should also include links to datasets and instructions on where to download to (for any data-sets too large to fit into the repo)

File: python/utils/combine_ru_data.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+        output_filename = os.path.join(ru_data_path, dataset, f"{dataset}_combined.txt")
+        with open(output_filename, "w") as outfile:
+            for pair in pairs:
+                outfile.writelines(process_pair(pair, dataset_dir))
+
+        print(f"Wrote all processed examples in {dataset} to {output_filename}.")
+
+
+if __name__ == "__main__":
```
Comment (ninalopatina) [inline python/utils/combine_ru_data.py]: this should be in a separate main.py file, with an import to all the functions in combine_ru_data

File: python/utils/spacy_eval.ipynb [hunk 4]
```diff
@@ -0,0 +1,169 @@
+{
```
Comment (ninalopatina) [inline python/utils/spacy_eval.ipynb]: this notebook should be in a notebooks folder in the root directory (along with the other notebooks in the exploratory_nbs folder); typically the python folder should only contain .py files.

File: python/utils/spacy_eval.ipynb [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+ "cells": [
```
Comment (ninalopatina) [inline python/utils/spacy_eval.ipynb]: this notebook should also include the installation instructions for this dataset as a comment: python -m spacy download en_core_web_lg

File: python/utils/spacy_eval.ipynb [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    "        y = [f, p, r]\n",
+    "    else: \n",
+    "        y = list(scores[tag].values())\n",
+    "    \n",
+    "    sns.barplot(x=x, y=y, ax=a, palette='GnBu', edgecolor='black')\n",
+    "    a.set_title(str.title(tag))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
```
Comment (ninalopatina) [inline python/utils/spacy_eval.ipynb]: This section is a great use of a jupyter notebook

File: python/utils/train_model.py [hunk 7]
```diff
@@ -184,82 +220,111 @@ def flat_accuracy(preds, labels):
...
 
-print('Initialized optimizer and set hyperparameters.')
+print("Initialized optimizer and set hyperparameters.")
 
 ###
 
+epoch = 0
 for _ in trange(EPOCHS, desc="Epoch"):
-
+    epoch = epoch + 1
```
Comment (ninalopatina) [inline python/utils/train_model.py]: I know I wrote this but.. should be "epoch += 1"

Comment (ninalopatina) [reply python/utils/train_model.py]: forgot to include this comment in the review: the train_model should be imported as a module, and the code below the functions that runs it should be in a separate main.py file

File: README.md (hunk 2 above)
Comment (chambliss) [reply README.md]: So far, none of the datasets are too large - down the line, should they be removed anyway and replaced with shell scripts to download them?

Comment (chambliss) [reply README.md]: notes from meeting: "This is a nice to have - not necessary for a small project, but does look good for production"

File: python/utils/train_model.py (hunk 7 above)
Comment (chambliss) [reply python/utils/train_model.py]: Nina from meeting: "not urgent, just one of those things; as I'm reviewing I have to comment, but I figured that was the case"

**Pull request merged**